### PR TITLE
Backend | ContentPage

### PR DESCRIPTION
**Pull request by engineervix**
This PR does the following:

1. Reorganises **models** in the `core` app, introducing a structure as follows:

   ```console
   .
   ├── models
   │   ├── __init__.py
   │   ├── mixins.py
   │   ├── pages.py
   │   └── snippets.py
   ```

   All the models previously in the `utils` app are moved to the `core` app, with the resulting migrations being modified as follows:

      - [`migrations.SeparateDatabaseAndState`](https://docs.djangoproject.com/en/3.2/ref/migration-operations/#django.db.migrations.operations.SeparateDatabaseAndState) is used to eliminate changes to the database
      - use [`migrations.AlterModelTable`](https://docs.djangoproject.com/en/3.2/ref/migration-operations/#altermodeltable) to rename tables
      - where applicable, set `db_operations` to an empty list to prevent Django from applying any changes to the database

2. Adds a new `ContentStoryBlock` consisting of blocks defined in #185 
3. Adds a new `ContentPage` model in the `core` app, whose `body` StreamField is the `ContentStoryBlock` above
4. Adds a new `HeroMixin`, which is used by the `ContentPage` above, and will be used by other `Page`s

Here are some screenshots:

image

image

### CODE REVIEW

File: wagtailio/core/blocks.py [hunk 1]
```diff
@@ -548,3 +548,25 @@ class Meta:
...
         template = "patterns/components/streamfields/text_and_media_block/text_and_media_block.html"
+
+
+class ContentStoryBlock(blocks.StreamBlock):
+    rich_text = RichTextBlock()
+    text_and_media = TextAndMediaBlock()
+    headline = HeadlineBlock()
+    highlight = HighlightBlock()
+    teaser = TeaserBlock()
+    icon_bullet = IconBulletBlock()
```
Comment (zerolab) [inline wagtailio/core/blocks.py:559]: I don't think there is ever a case where we just add the one icon bullet.

File: wagtailio/core/blocks.py [hunk 2]
```diff
@@ -548,3 +548,25 @@ class Meta:
...
+    icon_bullets = IconBulletsBlock()
+    card = CardBlock(group="Cards")
```
Comment (zerolab) [inline wagtailio/core/blocks.py:561]: similar to icon bullet. this is meant to go in a group of cards

File: wagtailio/core/blocks.py [hunk 3]
```diff
@@ -548,3 +548,25 @@ class Meta:
...
+    cards = CardsBlock(group="Cards")
+    logo_card = LogoCardBlock(group="Cards")
```
Comment (zerolab) [inline wagtailio/core/blocks.py:563]: similar to icon bullet. this is indended to be in a group of logo cards

File: wagtailio/utils/migrations/0002_auto_20220802_1706.py [hunk 4]
```diff
@@ -0,0 +1,83 @@
+# Generated by Django 3.2.12 on 2022-08-02 17:06
+
+from django.db import migrations
```
Comment (zerolab) [inline wagtailio/utils/migrations/0002_auto_20220802_1706.py:3]: general note: please, can we use named migrations?

File: wagtailio/core/models/snippets.py [hunk 5]
```diff
@@ -2,14 +2,8 @@
 
 from modelcluster.fields import ParentalKey
```
Comment (zerolab) [inline wagtailio/core/models/snippets.py:3]: the utils snippets are not related to content page. Can you please undo this change?
Since we will ultimately replace the existing menu relates snippets, the move is just unnecessary

File: wagtailio/utils/migrations/0002_auto_20220802_1706.py [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ("utils", "0001_initial"),
+    ]
+
+    operations = [
+        migrations.SeparateDatabaseAndState(
```
Comment (zerolab) [inline wagtailio/utils/migrations/0002_auto_20220802_1706.py:13]: praise: this is a cool use of the migrations functionality

alas, it is something that is out of scope of this feature

**Review by zerolab — CHANGES_REQUESTED** [review]
Lots of good stuff, @engineervix 

Left a number of issue to address

File: wagtailio/core/models/snippets.py (hunk 5 above)
Comment (engineervix) [reply wagtailio/core/models/snippets.py:3]: Thanks, will do :+1:

**Comment by engineervix** [comment]
Closing this PR and opening a new one with changes

**Comment by engineervix** [comment]
Superseded by #196
